### PR TITLE
SALTO-2087 Account id change validator

### PR DESCRIPTION
**Pull request by IdoZakk**
_Change Validators for account id_

---

Extends on the account id and display name filters PR  https://github.com/salto-io/salto/pull/3348 First two commits are not relevant here

---
_Release Notes_: 
Deploy with a wrong account id will fail with an appropriate error message

---
_User Notifications_: 
None

### CODE REVIEW

**Review by alonstern — COMMENTED** [review]
Looks good 👍 please link the PR this is built on and add to the description what commits need to be reviewed

File: packages/jira-adapter/src/change_validators/account_id.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      const idMap = await createIdToUserMap(paginator)
+      const changeErrors: ChangeError[] = []
+      changes
+        .filter(isAdditionOrModificationChange)
+        .filter(isInstanceChange)
+        .map(change => getChangeData(change))
+        .filter(isAccountIdType)
+        .forEach(element =>
+          walkOnElement({ element,
+            func: walkOnUsers(checkAndCreateChangeErrors(idMap, baseUrl, changeErrors)) }))
```
[the remark this answers is not in the document]
Comment (IdoZakk) [reply packages/jira-adapter/src/change_validators/account_id.ts:136]: I asked Lior in Jira about that

File: packages/jira-adapter/src/change_validators/account_id.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        return []
+      }
+      const { baseUrl } = client
+      const idMap = await createIdToUserMap(paginator)
+      const changeErrors: ChangeError[] = []
+      changes
+        .filter(isAdditionOrModificationChange)
+        .filter(isInstanceChange)
+        .map(change => getChangeData(change))
+        .filter(isAccountIdType)
```
[the remark this answers is not in the document]
Comment (IdoZakk) [reply packages/jira-adapter/src/change_validators/account_id.ts:133]: You mean the accountId type?

File: packages/jira-adapter/src/change_validators/account_id.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (IdoZakk) [reply packages/jira-adapter/src/change_validators/account_id.ts:136]: Per Lior's comment I will leave it as it is

**Review by alonstern — APPROVED** [review]
🚀 🚀

**Pull request merged**